Comp val (In progress)

Ciheim opened the pull request:
Early work on comprehensive validation

Commit:
Early Comp Val

Commit:
Adding ignore case + typo correction

Commit:
Revert " Adding ignore case + typo correction"

This reverts commit c456d2102aeb6c0ddcbc4e59c820fc0e662d18ca.

Commit:
Adjusting print statement

ceblanton commented on tests/compval:
```diff
@@ -0,0 +1,44 @@
+def main(json_path):
+
+    #Get/set cv path
+    cv_path = './CMIP6_frequency.json'
```
The cv path will be different for each vocabulary item, and it probably won't be called "CMIP6_frequency", and it won't be in the current directory.

We want to retrieve the vocabulary referenced in the schema. e.g.

https://github.com/aradhakrishnanGFDL/CatalogBuilder/blob/main/cats/gfdl_template.json#L22

But also the chunk_freq vocabulary:

https://github.com/aradhakrishnanGFDL/CatalogBuilder/blob/main/cats/gfdl_template.json#L50

ceblanton commented on tests/compval:
```diff
@@ -0,0 +1,44 @@
+    for x in cv["frequency"]:
+    #    print(x)
+        freq_cv.append(x)
+    #print(freq_cv)
```
This looks good but is hard-coded for frequency. We want it to work for other vocabulary too, e.g. chunk_freq

ceblanton reviewed (changes requested):
This looks like a great draft start that I admittedly haven't run yet.

The main limitation I see so far is retrieving the vocabulary values for the vocabulary items that require value validation should be more flexible. It shouldn't assume the vocabulary is a particular file name, in the local directory, and should work for other vocabulary types too (not just frequency).

Commit:
Compval (not done)

Ciheim committed:
Merge branch 'main' into CompVal

Commit:
Latest version of compval

Commit:
Merge remote-tracking branch 'refs/remotes/origin/CompVal' into CompVal

ceblanton commented on tests/compval:
```diff
@@ -0,0 +1,110 @@
+        for x in j["attributes"]:
+
+            #Checks to see if the vocabulary field is filled out. If so, use the vocab. Also, we avoid chunk_freq because it's currently broken.
+            if list(x.values())[1] and str(list(x.values())[0]) != 'chunk_freq':
```
I think it would be better to exit with error if the vocabulary can't be retrieved as the schema specifies.

If the http location in the schema does not exist, that is clearly a problem with the schema.

ceblanton reviewed (changes requested):
Can this functionality be added to scripts/test_catalog.py rather than a new tool?

aradhakrishnanGFDL commented:
Add test instructions with a documented test-case to this PR.

aradhakrishnanGFDL committed:
Update gfdl_template.json

removed chunk freq json which was an invalid url, dont want the catalog to be constrained of this column for now

Ciheim committed:
Merge branch 'main' into CompVal

Commit:
Catching failure cases

Commit:
Whitespace

Commit:
Instructions

Ciheim commented:
TEST CASES:

(Against GFDL CV's, adjusted to match CV's, expecting success)
Command: python compval ~/test_cat2.json 
Output: "Check passed."

(Against GFDL CV's, expecting helpful output)
Command: python compval ~/test_cat.json
Output: Found inconsistent value(s):['monthly', 'daily', 'annual']

(Against GFDL CV's, invalid JSON url, expecting failure)
Command: python compval ~/test_cat.json
Output: "Unable to open json: chunk_freq"

(Against GFDL CV's, adjusted to match CV's, expecting success)
Command: python compval ~/test_cat2.json /home/Ciheim.Brown/CMIP6_CVs
Output: "Check passed."

(Against CMIP CV's, expecting helpful output)
Command: python compval ~/test_cat.json /home/Ciheim.Brown/CMIP6_CVs
Output: "WARNING: NaN's found in: ['grid_label', 'institution_id', 'table_id'] Found inconsistent value(s):['dev', 'c96L65_am5f3b1r0_pdclim1850F', 'monthly', 'daily', 'annual', 'am5']"

(Against CMIP CV's, invalid JSON, expecting failure)
Command: python compval ~/test_cat.json /home/Ciheim.Brown/CMIP6_CVs
Output: Unable to open json: FakeVocabulary

Commit:
Space
